### PR TITLE
Fix Issue #632 - obj file section contents truncated

### PR DESCRIPTION
**Pull request by gmarkall**
Section contents were truncated at null bytes because `c_char_p` implies a null-terminated string. Swapping the return type of `LLVMPY_GetSectionContents` for a `POINTER(c_char)` no longer implies this.

Test case added based on the reproducer in #632.

### CODE REVIEW

**Comment by gmarkall** [comment]
The fail is a conda HTTP error - can someone re-run the build please?

**Comment by seibert** [comment]
I've kicked off re-running the failed jobs.

File: llvmlite/tests/test_binding.py [hunk 1]
```diff
@@ -173,6 +173,29 @@ def no_de_locale():
...
+    "6d746162003c737472696e673e00000000000000000000000000000000000000000000" \
+    "0000000000000000000000000000000000000000000000000000000000000000000000" \
+    "00000000000000001f0000000300000000000000000000000000000000000000a80000" \
+    "0000000000380000000000000000000000000000000100000000000000000000000000" \
+    "000001000000010000000600000000000000000000000000000040000000000000001f" \
+    "000000000000000000000000000000100000000000000000000000000000000f000000" \
+    "01000000000000000000000000000000000000005f0000000000000000000000000000" \
+    "0000000000000000000100000000000000000000000000000027000000020000000000" \
+    "0000000000000000000000000000600000000000000048000000000000000100000002" \
+    "00000008000000000000001800000000000000"
```
Comment (stuartarchibald) [inline llvmlite/tests/test_binding.py:192]: Can a comment please be added saying that this hex sequence on x86_64 linux produces:
```
$ objdump -D 623.elf 

623.elf:     file format elf64-x86-64


Disassembly of section .text:

0000000000000000 <__brybo>:
        ...
  10:   00 48 b9                add    %cl,-0x47(%rax)
  13:   01 20                   add    %esp,(%rax)
  15:   00 04 80                add    %al,(%rax,%rax,4)
  18:   00 10                   add    %dl,(%rax)
  1a:   70 48                   jo     64 <__brybo+0x64>
  1c:   0f af c8                imul   %eax,%ecx
```
just as a reference for the contents, thanks.

Comment (gmarkall) [reply llvmlite/tests/test_binding.py:192]: Sure!

Comment (gmarkall) [reply llvmlite/tests/test_binding.py:192]: I'm finding that my output is different to yours:

```
$ objdump -D 623.elf 

623.elf:     file format elf64-x86-64


Disassembly of section .text:

0000000000000000 <__arybo>:
   0:	48 c1 e2 20          	shl    $0x20,%rdx
   4:	48 09 c2             	or     %rax,%rdx
   7:	48 89 d0             	mov    %rdx,%rax
   a:	48 c1 c0 3d          	rol    $0x3d,%rax
   e:	48 31 d0             	xor    %rdx,%rax
  11:	48 b9 01 20 00 04 80 	movabs $0x7010008004002001,%rcx
  18:	00 10 70 
  1b:	48 0f af c8          	imul   %rax,%rcx
```

How did you produce `623.elf`? I used

```python
raw = bytes.fromhex(issue_632_elf)

with open("623.elf", "wb") as f:
    f.write(raw)
```

Comment (stuartarchibald) [reply llvmlite/tests/test_binding.py:192]: ah, I recall now, I manually modified the ELF when testing it, re-dumping the bytes gives the same as you!

Comment (gmarkall) [reply llvmlite/tests/test_binding.py:211]: Reference output now added.

Comment (stuartarchibald) [reply llvmlite/tests/test_binding.py:211]: Thanks!

**Review by stuartarchibald — APPROVED** [review]
Thanks for fixing this, patch looks good.

**Pull request merged**